helm/lint.go

```diff
@@ -22,80 +30,198 @@ type KubeObject struct {
+		if service.Spec.Type != "ExternalName" {
+			if service.Spec.Selector["release"] != release {
+				e := fmt.Errorf("Service with type '%v' and name '%v': object's spec.selector is missing the `release` "+
+					"key with the release name as a value (in this case, '%v'). Found these keys on spec.selector: %+v",
+					service.Spec.Type, service.ObjectMeta.Name, release, service.Spec.Selector)
 				errors = append(errors, e)
 			}
-			ctx.Logger.Debugf("Service with type '%v' and name '%v': object spec.selector exists, and the release key is %v", obj.Spec.Type, obj.Metadata.Name, obj.Spec.Selector["release"])
+			ctx.Logger.Debugf("Service with type '%v' and name '%v': object spec.selector exists, and the release "+
+				"key is %v", service.Spec.Type, service.ObjectMeta.Name, service.Spec.Selector["release"])
+		}
+	}
+
+	if ctx.Pedantic {
+		pedanticErrors := PedanticLintObject(ctx, obj)
+		if len(pedanticErrors) > 0 {
+			errors = append(errors, pedanticErrors...)
+		}
+	}
+
+	return errors
+}
+
+func PedanticLintObject(ctx *ankh.ExecutionContext, obj runtime.Object) []error {
+	errors := []error{}
+
+	switch obj.(type) {
+	case *appsv1beta1.Deployment:
+		deployment := obj.(*appsv1beta1.Deployment)
+
+		for _, c := range deployment.Spec.Template.Spec.Containers {
+
+			// Deployment images should only point to our production repositories
+			validRepo := false
+			repos := ctx.AnkhConfig.SupportedImageRepositories
+			if len(repos) > 0 {
+				for _, repo := range repos {
+					if strings.HasPrefix(c.Image, repo) {
+						ctx.Logger.Debugf("[Pedantic] Container '%v' in deployment '%v' has image '%v' "+
+							"correctly pointing to '%v'.", c.Name, deployment.ObjectMeta.Name, c.Image, repo)
+						validRepo = true
+						break
+					}
+				}
+
+				if !validRepo {
+					e := fmt.Errorf("[Pedantic] Container '%v' in deployment '%v' has image '%v'. Deployment "+
+						"image must point to one of %v.", c.Name, deployment.ObjectMeta.Name, c.Image, repos)
+					errors = append(errors, e)
+				}
+			}
+
+			// Deployment should omit imagePullPolicy or have it set to IfNotPresent since we use immutable images in production.
+			if c.ImagePullPolicy != "" && c.ImagePullPolicy != apiv1.PullIfNotPresent {
+				e := fmt.Errorf("[Pedantic] Container '%v' in deployment '%v' has pull policy '%v'. "+
+					"Pull policy should be set to '%v' or omitted.", c.Name, deployment.ObjectMeta.Name, c.ImagePullPolicy,
+					apiv1.PullIfNotPresent)
+				errors = append(errors, e)
+			} else {
+				ctx.Logger.Debugf("[Pedantic] Container '%v' in deployment '%v' has correctly omitted imagePolicy "+
+					"or set it to '%v'.", c.Name, deployment.Name, apiv1.PullIfNotPresent)
+			}
+
+			// Deployment should specify a readinessProbe
+			if c.ReadinessProbe == nil {
```

Comment:
After further thought and reading, I think the minimum should actually be a `livenessProbe` instead of a `readinessProbe`

Reply:
Yeah, makes sense